lib/Backends/OpenCL/kernels.cl

```diff
@@ -641,6 +695,37 @@ __kernel void matmulW(__global void *mem, cl_uint32_t dest, cl_uint32_t lhs,
   matmulK(&mem[dest], &mem[lhs], &mem[rhs], ddim, ldim, rdim);
 }
 
+__kernel void matmul_i8K(__global cl_int8_t *dest, __global cl_int8_t *lhs,
+                      __global cl_int8_t *rhs, ShapeNHWC ddim, ShapeNHWC ldim,
+                      ShapeNHWC rdim, cl_int32_t lhsOffset, cl_int32_t rhsOffset,
+                       cl_int32_t destOffset, cl_int32_t destPre, 
+                      cl_int32_t destPost, cl_int32_t destScale) {
+  // For each X in the destination matrix.
+  size_t x = get_global_id(0);
+  // For each Y in the destination matrix.
+  size_t y = get_global_id(1);
+
+  // Perform DOT on the row an column.
+  cl_int32_t sum = 0;
+  for (size_t i = 0; i < ldim.h; i++) {
+    sum += (lhs[getNHWC(ldim, x, i, 0, 0)] - lhsOffset) 
+            * (rhs[getNHWC(rdim, i, y, 0, 0)] - rhsOffset);
```

Comment:
Is it clang-formatted? Please run clang-format on the cpp files and on the opencl code you touched.

Reply:
formatted